course-design.md

```diff
@@ -0,0 +1,214 @@
+# Course design components for *RSE in R*
```

Comment:
I think it would be a minor nice-to-have if the title ends up being the same format as the Python book, but not necessary, especially not if the scope is a bit different.